qiskit/ignis/verification/randomized_benchmarking/dihedral.py

```diff
@@ -333,53 +338,53 @@ def __str__(self):
 
 class CNOTDihedral():
     """CNOT-dihedral Object Class.
-    The CNOT-dihedral group on n qubits is generated by the gates
+    The CNOT-dihedral group on num_qubits qubits is generated by the gates
     CNOT, T and X."""
 
-    def __init__(self, n_qubits):
-        # Construct the identity element on n_qubits qubits.
-        self.n_qubits = n_qubits
+    def __init__(self, num_qubits):
+        # Construct the identity element on num_qubits qubits.
+        self.num_qubits = num_qubits
         # phase polynomial
-        self.poly = SpecialPolynomial(n_qubits)
+        self.poly = SpecialPolynomial(num_qubits)
         # n x n invertible matrix over Z_2
-        self.linear = [[int(r == c) for c in range(n_qubits)]
-                       for r in range(n_qubits)]
+        self.linear = [[int(r == c) for c in range(num_qubits)]
+                       for r in range(num_qubits)]
         # binary shift, n coefficients in Z_2
-        self.shift = n_qubits * [0]
+        self.shift = num_qubits * [0]
 
     def _z2matmul(self, left, right):
         """Compute product of two n x n z2 matrices."""
-        prod = [[0 for col in range(self.n_qubits)]
-                for row in range(self.n_qubits)]
-        for i in range(self.n_qubits):
-            for j in range(self.n_qubits):
-                for k in range(self.n_qubits):
+        prod = [[0 for col in range(self.num_qubits)]
```

Comment:
Python for loops are a very slow way to do this. If the input are numpy arrays this should be able to be done like. `np.mod(np.dot(left, right), 2)`